config/routes.rb

```diff
@@ -1,6 +1,15 @@
 Rails.application.routes.draw do
+  post "oauth/callback" => "oauths#callback"
+  get "oauth/callback" => "oauths#callback"
+  get "oauth/:provider" => "oauths#oauth", :as => :auth_at_provider
+
   resources :cards
+  resources :sessions
+  resources :users
 
   root "home#index"
-  put "review_card" => "home#review_card"
+  get "login" => "sessions#new"
+  get "review" => "review#index"
+  put "review_card" => "review#review_card"
+  post 'logout' => 'sessions#destroy', :as => :logout
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.